### DISCUSSION TITLE
Presets for larger multirotors

### DISCUSSION THREAD

**Discussion by RCnator** [iNavFlight/inav#9765] [Ideas]
## Current Behavior

Currently there is only presets for quads up to 7 inch.
## Desired Behavior

Since I guess the name Inav stands autonomous flight maybe there should be presets for larger quads and not only racing size quads. In my case 15 inch.
## Suggested Solution

Maybe add presets from 9 inch with 3 inch increments up to 18?

## Who does this impact? Who is this for?

Quad and multirotor users, all skill levels.

**Comment by intelligentpotato** [comment]
+1 on this.
I believe iNav 5 had a preset for 10" which was a fine starting point for everything big and slow, but they removed it in iNav 6. Additionally, it seems to me that they have changed the PID controller logic/sensitivity in iNav 6. I flashed iNav 6 yesterday, migrated the iNav 5 settings that flew fine and crashed upon take off by flipping over. The thrust graph for that "flight" looks like god damn saw teeth. My PIDs are 90 34 54 - basically the old defaults.

**Comment by MrD-RC** [comment]
You’re more than welcome to suggest actual settings that work with a range of different quad at those safe settings.

**Comment by RCnator** [comment]
Unfortunately I myself lack the skill and know how to do that but hopefully someone else picks up the ball.

**Comment by shota3527** [comment]
> You’re more than welcome to suggest actual settings that work with a range of different quad at those safe settings.

Agreed, we need a safe settings for wide range of quads.

**Comment by agoliveira** [comment]
I would like to leave my +1 here. I just added a 10" to my fleet and I'm having a very hard time trying to make it work. It oscillates a lot. The funny thing is that I used inav 2.51 IIRC in a very large quad and worked quite well.

**Comment by Dronitoni** [comment]
I currently try to find the best PID's for my 14-inch Quad. Seems not that easy in INAV6. 
This 14-incher is an old one which I was flying with a DJI-Naza controller in the early days.
Now I switched this Quad to a Matek H743 and the tuning work begins again.

**Reply by tims5833** [reply]
This is a quite old discussion and i am new to iNav but here goes. I have a 22" quad which had Matek H743 installed by someone who tried to help me get it going. Sadly it is unstable on takeoff and I'm looking for suggestions for PIDs to try ?

**Reply by tims5833** [reply]
ChatGPT came up with some really low numbers for the PIDs so I'm going to give them a whirl

**Reply by sensei-hacker** [reply]
You might want to carefully sanity check anything that comes from a predictive text, before applying it on a dangerous machine that is going to be flying around people.



![image](https://github.com/user-attachments/assets/1edc81a8-fa32-4f31-a011-c4c0c00a4997)

![chatgpt-fail-09](https://github.com/user-attachments/assets/9db5ad26-815c-4b7a-b5d7-14b992b1f1b5)

![letterY](https://github.com/user-attachments/assets/a195b8a0-b5f7-4ea3-a902-02bf6521b541)

**Reply by tims5833** [reply]
I have the machine chained to the earth for testing so risk is zero save to the propeller blade that got damaged in the attempt - I am now going to look at alternatives to Inav, any suggestions ?

**Reply by sensei-hacker** [reply]
> I have the machine chained to the earth for testing so risk is zero save to the propeller blade that got damaged in the attempt - I am now going to look at alternatives to Inav, any suggestions ?

I would suggest that you start with the settings that have worked well for others, such as Ivan. Be aware the tethers will affect the flight, particularly if they aren't very light.

**Comment by RCnator** [comment]
I can see Inav 7 is out now but there is no mentioning of improvements regarding this issue unfortunately.

**Comment by MrD-RC** [comment]
> I can see Inav 7 is out now but there is no mentioning of improvements regarding this issue unfortunately.

No one has put forward any suggested setups. AFAIK, no developers have quads over 7”. As already stated, someone with larger quads will need to make suggested settings which are tested by the community and work on a wide range of quads with the target prop size.

TBH, there just doesn’t seem to be that much interest. Maybe start a conversation on the INAV discord’s multi-rotor channel?

**Comment by MrD-RC** [comment]
I created the discussion on Discord. https://discord.com/channels/791437907478577172/1173188186382549032

**Comment by petrweigl** [comment]
is there some progress on this issue?

**Comment by MrD-RC** [comment]
There had been no preset recommendations. Also not a lot of interest on Discord. It seems like most people don’t fly quads that size.

**Comment by petrweigl** [comment]
I have quadcopter with 12inch props running on inav7, trying to tune it. I am getting poor performance … with my settings.

**Comment by RCnator** [comment]
A lot of us dont have a discord account, see above, and dont want to open up yet another online account, especially when we already have github. 

For Inav to live up to its name I think there should be a bit more focus on larger multirotors rather than racing quads, but I also remember when we were contested requesting dshot, [https://github.com/iNavFlight/inav/issues/1489]

**Comment by MrD-RC** [comment]
The problem is there is simply not enough interest. And the INAV pilots who actually have these larger quads aren’t coming up with any suggestions for settings.  These settings will need to work for a large range of quads in that size category. 

So without any suggestions for settings, which are then tested by the community. The profiles cannot be added. If you guys with 10” quads want a preset. I’m sorry, but you’ll need to put in the legwork, experiment with the tunes, share findings with others to help come up with the settings. Without that. The presets you want won’t happen. The ball is in your court guys. 

I’m going to convert this to a discussion. As that is now more fitting.

**Comment by Bloody-Badboy** [comment]
+1 on this. I have a quad with TBS Discovery frame 2213 935kv Motor 10" Props. Previously I was using Pixhawk flight controller and it was stable and smooth, now I am running Inav 7 but am not able to tune it. It oscillates too much I have been trying to tune it for the last couple of days but have not been able to.  🥲🥲 No luck. 

I compiled nav 2.4.0 installed it on the board and applied the default 10" preset it's pretty much more stable and more flyable.

**Comment by Ivan29777** [comment]
Only problem with quads from 7 inch are vibrations. But my drons fly indoor for testing from 7 up to 16 inch. They are very stable. You need to know what you doing. Vibrations, filters, PID.  Use Inav blackox log and pidtoolox. Drons from 7 inch are hard to tune. Some nice starting PID 55 75 35, if vibrations are not huge go for 60 75 40. Gyro filters in my 16 inch  in 50hz all tuned in Inav 7.1.  On the drons from 7 inch my flight controlers are in some 3d printed box. Moving air is so big. Why n3 and a3 have fc in box?.  Good starting point for 10 inch is 7 inch setup from pawel only lower filters by 10hz. My fc its not on esc like you see aio tower... Its solo down centre of drone. Esc is back or upper frome fc. Thats how need to by and if you want to fly big dron you will need 3d printed parts. Or buy a3 or pixhavk

**Reply by Ivan29777** [reply]
Setup: Tarot 650 quadcopter, FC kakute f405 with inav 7.1.2., esc mamba 4in1 50a, motors 4114 400kv 16 inch prop. For testing I use 4s batteri 5200mah
Tarot frame was moded to lower vibrations: 
3d printed links from motor to motor( 4 links) PLA
ESC is back and fc is lower the centre in 3d printed box. I use TPU 85 infill 100% for box
tpu dampers on motors and screws
My log file is here if someone help:  Need some tune for PID but fly is good and stable indoor propwash test
[blackbox_log_2024-09-12_115002.TXT](https://github.com/user-attachments/files/16977436/blackbox_log_2024-09-12_115002.TXT)

next project is 10 inch full printed and fly like 7 inch... I will add log for that BEAST. Project was 1 year old but i will update all and post logs....  Matek f722se, 35a esc, 10 inch props, fully printed only hands aluminium... compas with gps bn880 
In this setup esc is up and fc down centre isolated. In all my project i use overkill of 2 1000 35v capacitors

**Reply by tims5833** [reply]
I have these PIDS from the 7" setup and am struggling with my 22" quad
Inav 7” blades
Roll 	 40:30:23:60
Pitch 40:30:23:60
Yaw	 85:45:  0:60
Can you suggest a starting point for such a large drone ?

**Reply by Dronitoni** [reply]
![INAV7_14-incher](https://github.com/user-attachments/assets/8b74d4a5-2f66-4d09-a5ae-3d59c8b77dc5)

I'm using this on my 14-incher.

**Reply by Dronitoni** [reply]
INAV 7

**Comment by tims5833** [comment]
Hi Sensai, the tethers were chains and yes they were not light but boy has
this drone got some lifting power with 4 x kv170 motors and 12S battery
power. Can you point me to Ivan's settings (I couldnt find them when I
looked through the discussion)
Brgds,
Tim

**Comment by tims5833** [comment]
Hi Sensai, the tethers were chains and yes they were not light but boy has
this drone got some lifting power with 4 x kv170 motors and 12S battery
power. Can you point me to Ivan's settings (I couldnt find them when I
looked through the discussion)
Brgds,
Tim

**Reply by Qau44** [reply]
Hi Tim, I'm having the same issue with my drone and I'm using 4 x kv340 with 22'' props and I'm getting major oscillations. I've been trying to deal with it settings for the past 2 months now but of course there's not much info on larger drones sadly. i was getting pretty close using betaflight and though that inav would've been better but honestly i feel like I'm back at square one again.

**Comment by sensei-hacker** [comment]
@tims5833  Ivan wrote above:

Some nice starting PID 55 75 35, if vibrations are not huge go for 60 75 40. 

But really, trying to tune pids while it's chained down with heavy chains is likely to go badly.

**Comment by petrweigl** [comment]
i am using roll, pitch PID 55-30-48 for  15 inch and works nice. Its not critically dumped yet, but good enough.

**Comment by tims5833** [comment]
Hi Dronitoni and Sensei, the chains aren't really heavy, say about 1 Ib
each, but I will take your collective advice and find some nylon cord for
the next attempt (though it broke 4 aluminium carabiners on the end of the
chains). I had the Mateksys H743 V3 flight controller and Inav 7.1
installed by a local "expert" who then crashed it a couple of times so I am
being as cautious as I can. If you're on WhatsApp I could send you a video
of the next attempt which hopefully will at least be worth a giggle. Thank
you for your help.
Brgds,
Tim +64 212161770

**Comment by Ivan29777** [comment]
Here is for 16 inch dron 

H Firmware type:Cleanflight
H Firmware revision:INAV 7.1.2 (4e1e59eb) KAKUTEF4V2
H Firmware date:May 31 2024 23:08:25
H Log start datetime:2041-06-28T01:04:00.000+00:00
H Craft name:
H P interval:1/2
H minthrottle:1127
H maxthrottle:1850
H gyro_scale:0x3f800000
H motorOutput:1127,1850
H acc_1G:2048
H vbat_scale:110
H vbatcellvoltage:33,35,42
H vbatref:1667
H currentMeter:0,400
H looptime:1000
H rc_rate:100
H rc_expo:40
H rc_yaw_expo:40
H thr_mid:50
H thr_expo:0
H tpa_rate:0
H tpa_breakpoint:1500
H rates:30,30,20
H rollPID:44,45,36,60
H pitchPID:44,45,36,60
H yawPID:50,45,0,60
H altPID:50,0,0
H posPID:65,0,0
H posrPID:40,15,100
H levelPID:20,15,75
H magPID:60
H velPID:100,50,10
H yaw_lpf_hz:0
H dterm_lpf_hz:40
H dterm_lpf_type:2
H deadband:4
H yaw_deadband:4
H gyro_lpf:0
H gyro_lpf_hz:50
H gyro_lpf_type:0
H dynamicGyroNotchQ:250
H dynamicGyroNotchMinHz:40
H acc_lpf_hz:15
H acc_hardware:3
H baro_hardware:4
H mag_hardware:0
H serialrx_provider:2
H motor_pwm_protocol:5
H motor_pwm_rate:8000
H debug_mode:0
H features:810028038
H waypoints:0,0
H acc_notch_hz:0
H acc_notch_cutoff:1
H pidSumLimit:500
H pidSumLimitYaw:350
H axisAccelerationLimitYaw:10000
H axisAccelerationLimitRollPitch:0
H rpm_gyro_filter_enabled:0
H rpm_gyro_harmonics:1
H rpm_gyro_min_hz:100
H rpm_gyro_q:500

**Reply by tims5833** [reply]
Thank you for the info Ivan. It may be that my unstable hovering issue may be down to motor/ESC calibration rather than PIDs as when flying on the ground with no propellers fitted I am finding the throttle stick does not control RPM exactly. Once the throttle is increased from idle the motors seem to have an arbitrary element where the revs rise and fall even when the stick isnt moved. I'm only going on sound at this point but it does seem that the TX is not in control of the motors and pulling back to idle isnt always successful. 
Any thoughts ?

**Reply by sensei-hacker** [reply]
>  the throttle stick does not control RPM exactly

I don't think multirotors have any flight mode where the pilot controls the RPM of each motor.

What flight mode are you in?

**Reply by tims5833** [reply]
These 2 pics are from a screenshot of the inav mode settings on my drone. I didnt set them so I'm not sure how to critique them:
![inav flight modes](https://github.com/user-attachments/assets/4a4bf800-ee61-4820-b25d-8d8a0c701314)
![inav flight mode modifiers](https://github.com/user-attachments/assets/cd29a24b-2873-4c84-8baa-1eca0b6b3b9d)

**Reply by tims5833** [reply]
Hi Sensei, when touching nothing but the left stick (fore and aft movement only) I had expected all 4 motors to increase and fall in revs together as I was asking it to gain and lose altitude. But they dont seem to be working together and when I move the stick to the bottom the revs dont fall back to idle so I am having trouble landing even though the drone is tethered.

**Reply by sensei-hacker** [reply]
I see you have channel 8 set up to switch between three different flight modes. But you don't know which mode you were in?

Suppose you were in Angle mode. In which case centering the pitch, roll axis would be commanding INAV to level out the aircraft - to spin up the motors on the side that is 1mm lower than the other side.

If the low side had a table attached as cargo, it's going to have to spin the motors really high to lift the table a little bit to level out.

That's actually a good way to think about it to keep things clear. Whatever you have attached to the drone is cargo, a payload. You're asking it lift the table or the building or whatever is attached. Along with the chains or ropes, of course.

**Reply by tims5833** [reply]
CH8 was in ANGLE mode and I have not changed it as yet - so far I've only used CH5 to ARM and the LH stick for revs. I hadnt realised a 1 deg offset on the ground due to slope would matter as I assumed it'd sort itself as soon as it lifted off. But I will check by getting it level and seeing what Inav sees then. At this point I'm thinking the ESCs might need calibrating as I saw that mentioned in a youtube video but as yet I haven't found the instructions on how to do this in Inav.
After that I was wondering whether I need to thoroughly check out the motors for their impedances etc. Also I have 4 electronic spring balances and was thinking I might try measuring the thrusts of all four motors ... any more thoughts would be gratefully received.

**Reply by Ivan29777** [reply]
> Thank you for the info Ivan. It may be that my unstable hovering issue may be down to motor/ESC calibration rather than PIDs as when flying on the ground with no propellers fitted I am finding the throttle stick does not control RPM exactly. Once the throttle is increased from idle the motors seem to have an arbitrary element where the revs rise and fall even when the stick isnt moved. I'm only going on sound at this point but it does seem that the TX is not in control of the motors and pulling back to idle isnt always successful. Any thoughts ?

Without propellers dron goes wild because you gave it throttle in Angle mode. In blheli you have to adjust it for a large drone.  stick comand its not problem if you use dshot protocol.  Can you post your log file. Check if you have adjusted the motor pole correctly. Lower the motor idle power. But again I say if you test without propellers in angle mode it is normal that the drone does not know what to do.  Check the rotation of the motor and propeller.

**Reply by tims5833** [reply]
I'm sure the motor rotation is correct and props also. UBLOX protocol has been set for navigation and STANDARD for ESC - what should I use DSHOT protocol for?
We are having wild wet weather after months of drought so will put up a logfile as soon as I can fly and work out how to retrieve the file from the "onboard SD card"

**Reply by sensei-hacker** [reply]
> I hadnt realised a 1 deg offset on the ground due to slope would matter as I assumed it'd sort itself as soon as it lifted off.

It'll be sorted out BEFORE the high side lifts off. The low side will lift off first. By spinning the motors on the low side faster.

If course, if there are no props on those motors, the ones on the low side will just spin faster and faster, trying to lift that side, but never succeeding.

>  But I will check by getting it level and seeing what Inav sees then.

As I recall, it uses 0.1° increments internally. I may mistaken. But yeah when you command it to level out, it will keep trying to level out until it's at 0.0°.

The bigger picture here is you can't test flight characteristics without flying.

**Reply by Ivan29777** [reply]
Watch a few videos on FPV University https://www.youtube.com/watch?v=phafJYNzTR0

**Comment by Ivan29777** [comment]
[blackbox_log_2024-09-21_162042.TXT](https://github.com/user-attachments/files/19523089/blackbox_log_2024-09-21_162042.TXT)

**Comment by agoliveira** [comment]
I've been working on a project that may be able to help, https://github.com/agoliveira/INAV-Toolkit
